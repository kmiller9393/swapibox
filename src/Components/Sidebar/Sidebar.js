import React, { Component } from 'react';

export default class Sidebar extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      episode: null,
      quote: ''
    };
  }
  componentDidMount = () => {
    fetch(`https://swapi.co/api/films/${Math.floor(Math.random() * 7) + 1}`)
      .then(response => response.json())
      .then(response => {
        const quote = response.opening_crawl;
        const title = response.title;
        const episode = response.episode_id;
        this.setState({ title, episode, quote });
      });
  };

  render() {
    return (
      <div className="sidebar">
        <div className="fade" />
        <section className="star-wars">
          <div className="crawl">
            <div className="title">
              <p>Episode {this.state.episode}</p>
              <h1>{this.state.title}</h1>
            </div>
            <p className="quote">{this.state.quote}</p>
          </div>
        </section>
      </div>
    );
  }
}
