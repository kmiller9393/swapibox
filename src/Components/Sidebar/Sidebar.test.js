import React from 'react';
import ReactDOM from 'react-dom';
import Sidebar from './Sidebar';
import { shallow } from 'enzyme';
import { mockMovies } from '../../mockData';

describe('Sidebar', () => {
  let wrapper;

  beforeEach(() => {
    window.fetch = jest
      .fn()
      .mockImplementation(() =>
        Promise.resolve({ json: () => Promise.resolve(mockMovies) })
      );
    wrapper = shallow(<Sidebar />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there is no quote', () => {
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve({
            opening_crawl: '',
            title: '',
            episode_id: '',
            release_date: ''
          })
      })
    );
    expect(wrapper).toMatchSnapshot();
  });
});
