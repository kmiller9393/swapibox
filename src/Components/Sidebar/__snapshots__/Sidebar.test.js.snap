// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <section
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <p>
                                        Episode 
                                        6
                              </p>
                              <h1>
                                        Return of the Jedi
                              </h1>
                    </div>
                    <p
                              className="quote"
                    >
                              Luke Skywalker has returned to
                              his home planet of Tatooine in
                              an attempt to rescue his
                              friend Han Solo from the
                              clutches of the vile gangster
                              Jabba the Hutt.
                              
                              Little does Luke know that the
                              GALACTIC EMPIRE has secretly
                              begun construction on a new
                              armored space station even
                              more powerful than the first
                              dreaded Death Star.
                              
                              When completed, this ultimate
                              weapon will spell certain doom
                              for the small band of rebels
                              struggling to restore freedom
                              to the galaxy...
                    </p>
                    <p>
                              1983-05-25
                    </p>
          </div>
</section>,
      ],
      "className": "sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <p>
                                    Episode 
                                    6
                        </p>
                        <h1>
                                    Return of the Jedi
                        </h1>
            </div>
            <p
                        className="quote"
            >
                        Luke Skywalker has returned to
                        his home planet of Tatooine in
                        an attempt to rescue his
                        friend Han Solo from the
                        clutches of the vile gangster
                        Jabba the Hutt.
                        
                        Little does Luke know that the
                        GALACTIC EMPIRE has secretly
                        begun construction on a new
                        armored space station even
                        more powerful than the first
                        dreaded Death Star.
                        
                        When completed, this ultimate
                        weapon will spell certain doom
                        for the small band of rebels
                        struggling to restore freedom
                        to the galaxy...
            </p>
            <p>
                        1983-05-25
            </p>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <p>
                                Episode 
                                6
                </p>
                <h1>
                                Return of the Jedi
                </h1>
</div>,
              <p
                className="quote"
>
                Luke Skywalker has returned to
                his home planet of Tatooine in
                an attempt to rescue his
                friend Han Solo from the
                clutches of the vile gangster
                Jabba the Hutt.
                
                Little does Luke know that the
                GALACTIC EMPIRE has secretly
                begun construction on a new
                armored space station even
                more powerful than the first
                dreaded Death Star.
                
                When completed, this ultimate
                weapon will spell certain doom
                for the small band of rebels
                struggling to restore freedom
                to the galaxy...
</p>,
              <p>
                1983-05-25
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Episode 
                    6
</p>,
                  <h1>
                    Return of the Jedi
</h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      6,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    6,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Return of the Jedi",
                  },
                  "ref": null,
                  "rendered": "Return of the Jedi",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...",
                "className": "quote",
              },
              "ref": null,
              "rendered": "Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1983-05-25",
              },
              "ref": null,
              "rendered": "1983-05-25",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <section
            className="star-wars"
>
            <div
                        className="crawl"
            >
                        <div
                                    className="title"
                        >
                                    <p>
                                                Episode 
                                                6
                                    </p>
                                    <h1>
                                                Return of the Jedi
                                    </h1>
                        </div>
                        <p
                                    className="quote"
                        >
                                    Luke Skywalker has returned to
                                    his home planet of Tatooine in
                                    an attempt to rescue his
                                    friend Han Solo from the
                                    clutches of the vile gangster
                                    Jabba the Hutt.
                                    
                                    Little does Luke know that the
                                    GALACTIC EMPIRE has secretly
                                    begun construction on a new
                                    armored space station even
                                    more powerful than the first
                                    dreaded Death Star.
                                    
                                    When completed, this ultimate
                                    weapon will spell certain doom
                                    for the small band of rebels
                                    struggling to restore freedom
                                    to the galaxy...
                        </p>
                        <p>
                                    1983-05-25
                        </p>
            </div>
</section>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <div
                            className="title"
              >
                            <p>
                                          Episode 
                                          6
                            </p>
                            <h1>
                                          Return of the Jedi
                            </h1>
              </div>
              <p
                            className="quote"
              >
                            Luke Skywalker has returned to
                            his home planet of Tatooine in
                            an attempt to rescue his
                            friend Han Solo from the
                            clutches of the vile gangster
                            Jabba the Hutt.
                            
                            Little does Luke know that the
                            GALACTIC EMPIRE has secretly
                            begun construction on a new
                            armored space station even
                            more powerful than the first
                            dreaded Death Star.
                            
                            When completed, this ultimate
                            weapon will spell certain doom
                            for the small band of rebels
                            struggling to restore freedom
                            to the galaxy...
              </p>
              <p>
                            1983-05-25
              </p>
</div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  <p>
                                    Episode 
                                    6
                  </p>
                  <h1>
                                    Return of the Jedi
                  </h1>
</div>,
                <p
                  className="quote"
>
                  Luke Skywalker has returned to
                  his home planet of Tatooine in
                  an attempt to rescue his
                  friend Han Solo from the
                  clutches of the vile gangster
                  Jabba the Hutt.
                  
                  Little does Luke know that the
                  GALACTIC EMPIRE has secretly
                  begun construction on a new
                  armored space station even
                  more powerful than the first
                  dreaded Death Star.
                  
                  When completed, this ultimate
                  weapon will spell certain doom
                  for the small band of rebels
                  struggling to restore freedom
                  to the galaxy...
</p>,
                <p>
                  1983-05-25
</p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Episode 
                      6
</p>,
                    <h1>
                      Return of the Jedi
</h1>,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Episode ",
                        6,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Episode ",
                      6,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Return of the Jedi",
                    },
                    "ref": null,
                    "rendered": "Return of the Jedi",
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...",
                  "className": "quote",
                },
                "ref": null,
                "rendered": "Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1983-05-25",
                },
                "ref": null,
                "rendered": "1983-05-25",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Sidebar should match the snapshot when there is no quote 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <section
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <p>
                                        Episode 
                                        6
                              </p>
                              <h1>
                                        Return of the Jedi
                              </h1>
                    </div>
                    <p
                              className="quote"
                    >
                              Luke Skywalker has returned to
                              his home planet of Tatooine in
                              an attempt to rescue his
                              friend Han Solo from the
                              clutches of the vile gangster
                              Jabba the Hutt.
                              
                              Little does Luke know that the
                              GALACTIC EMPIRE has secretly
                              begun construction on a new
                              armored space station even
                              more powerful than the first
                              dreaded Death Star.
                              
                              When completed, this ultimate
                              weapon will spell certain doom
                              for the small band of rebels
                              struggling to restore freedom
                              to the galaxy...
                    </p>
                    <p>
                              1983-05-25
                    </p>
          </div>
</section>,
      ],
      "className": "sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <p>
                                    Episode 
                                    6
                        </p>
                        <h1>
                                    Return of the Jedi
                        </h1>
            </div>
            <p
                        className="quote"
            >
                        Luke Skywalker has returned to
                        his home planet of Tatooine in
                        an attempt to rescue his
                        friend Han Solo from the
                        clutches of the vile gangster
                        Jabba the Hutt.
                        
                        Little does Luke know that the
                        GALACTIC EMPIRE has secretly
                        begun construction on a new
                        armored space station even
                        more powerful than the first
                        dreaded Death Star.
                        
                        When completed, this ultimate
                        weapon will spell certain doom
                        for the small band of rebels
                        struggling to restore freedom
                        to the galaxy...
            </p>
            <p>
                        1983-05-25
            </p>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <p>
                                Episode 
                                6
                </p>
                <h1>
                                Return of the Jedi
                </h1>
</div>,
              <p
                className="quote"
>
                Luke Skywalker has returned to
                his home planet of Tatooine in
                an attempt to rescue his
                friend Han Solo from the
                clutches of the vile gangster
                Jabba the Hutt.
                
                Little does Luke know that the
                GALACTIC EMPIRE has secretly
                begun construction on a new
                armored space station even
                more powerful than the first
                dreaded Death Star.
                
                When completed, this ultimate
                weapon will spell certain doom
                for the small band of rebels
                struggling to restore freedom
                to the galaxy...
</p>,
              <p>
                1983-05-25
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Episode 
                    6
</p>,
                  <h1>
                    Return of the Jedi
</h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      6,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    6,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Return of the Jedi",
                  },
                  "ref": null,
                  "rendered": "Return of the Jedi",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...",
                "className": "quote",
              },
              "ref": null,
              "rendered": "Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1983-05-25",
              },
              "ref": null,
              "rendered": "1983-05-25",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <section
            className="star-wars"
>
            <div
                        className="crawl"
            >
                        <div
                                    className="title"
                        >
                                    <p>
                                                Episode 
                                                6
                                    </p>
                                    <h1>
                                                Return of the Jedi
                                    </h1>
                        </div>
                        <p
                                    className="quote"
                        >
                                    Luke Skywalker has returned to
                                    his home planet of Tatooine in
                                    an attempt to rescue his
                                    friend Han Solo from the
                                    clutches of the vile gangster
                                    Jabba the Hutt.
                                    
                                    Little does Luke know that the
                                    GALACTIC EMPIRE has secretly
                                    begun construction on a new
                                    armored space station even
                                    more powerful than the first
                                    dreaded Death Star.
                                    
                                    When completed, this ultimate
                                    weapon will spell certain doom
                                    for the small band of rebels
                                    struggling to restore freedom
                                    to the galaxy...
                        </p>
                        <p>
                                    1983-05-25
                        </p>
            </div>
</section>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <div
                            className="title"
              >
                            <p>
                                          Episode 
                                          6
                            </p>
                            <h1>
                                          Return of the Jedi
                            </h1>
              </div>
              <p
                            className="quote"
              >
                            Luke Skywalker has returned to
                            his home planet of Tatooine in
                            an attempt to rescue his
                            friend Han Solo from the
                            clutches of the vile gangster
                            Jabba the Hutt.
                            
                            Little does Luke know that the
                            GALACTIC EMPIRE has secretly
                            begun construction on a new
                            armored space station even
                            more powerful than the first
                            dreaded Death Star.
                            
                            When completed, this ultimate
                            weapon will spell certain doom
                            for the small band of rebels
                            struggling to restore freedom
                            to the galaxy...
              </p>
              <p>
                            1983-05-25
              </p>
</div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  <p>
                                    Episode 
                                    6
                  </p>
                  <h1>
                                    Return of the Jedi
                  </h1>
</div>,
                <p
                  className="quote"
>
                  Luke Skywalker has returned to
                  his home planet of Tatooine in
                  an attempt to rescue his
                  friend Han Solo from the
                  clutches of the vile gangster
                  Jabba the Hutt.
                  
                  Little does Luke know that the
                  GALACTIC EMPIRE has secretly
                  begun construction on a new
                  armored space station even
                  more powerful than the first
                  dreaded Death Star.
                  
                  When completed, this ultimate
                  weapon will spell certain doom
                  for the small band of rebels
                  struggling to restore freedom
                  to the galaxy...
</p>,
                <p>
                  1983-05-25
</p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Episode 
                      6
</p>,
                    <h1>
                      Return of the Jedi
</h1>,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Episode ",
                        6,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Episode ",
                      6,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Return of the Jedi",
                    },
                    "ref": null,
                    "rendered": "Return of the Jedi",
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...",
                  "className": "quote",
                },
                "ref": null,
                "rendered": "Luke Skywalker has returned to
his home planet of Tatooine in
an attempt to rescue his
friend Han Solo from the
clutches of the vile gangster
Jabba the Hutt.

Little does Luke know that the
GALACTIC EMPIRE has secretly
begun construction on a new
armored space station even
more powerful than the first
dreaded Death Star.

When completed, this ultimate
weapon will spell certain doom
for the small band of rebels
struggling to restore freedom
to the galaxy...",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1983-05-25",
                },
                "ref": null,
                "rendered": "1983-05-25",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
